### DISCUSSION TITLE
Final agreement on the default UI's tech & approach

### DISCUSSION THREAD

**Discussion by ghys** [openhab/openhab-webui#1810] [WebUI Maintainer Discussions]
Dear @openhab/webui-maintainers,

To wrap up the discussion in https://github.com/openhab/openhab-webui/issues/24, I'd like to formally call for a final agreement on the default UI for openHAB 3 - the tech, the feature set and the look & feel.
I've already stated my preferences in https://github.com/openhab/openhab-webui/issues/24, so no need to reiterate them here. I'm volunteering to dedicate some time in the next days/weeks to continue working on my initial prototype at https://github.com/ghys/openhab-ui-f7 and:
- Refactor, clean up & improve the existing "proof-of-concept grade" code:
  * Add an abstraction layer for API calls, SSE, because there could be problems with iOS/Android Cordova apps which I'll try to keep in the intended scope of this UI - I can get into details and discuss that elsewhere.
  * Introduce a Vuex store for state management. All other choices will be made in line with the consensus reached in https://github.com/openhab/openhab-webui/issues/24. I'm not sure anymore a global SSE connection listening to `rest/events` is strictly necessary for the entire duration of the session though. The event bus can be quite busy and experiences with HABot have proven this approach to be hogging CPUs and draining batteries for larger installations. Again this would be discussed in separate issues.
-  (Try to) add testing tools (Jest, Cypress) to the environment.
- Add the missing pages and functionality to reach a working version.
- Provide proposals for the "home page" - there is a expandable card proof-of-concept at https://habot-demo.azurewebsites.net/ with 3 (partly working, the cards are dynamically created but the controls are inoperative!) tabs: Locations, Equipments, Properties to slice the semantically tagged items into expandable cards. This card system could be enhanced with "favorites" and contextual ("Now") cards in the Overview tab, as well as buttons to activate Scenes (sequence of commands). Ideally card headers and maybe the "mini sitemap" displayed when expanding it could also be lightly customized by the user from the UI. The simple HABot client proof-of-concept at the top of the Overview tab will appear if the HABot bundle is installed. All these are completely new proposals and are not completely devised so I'll open another issue to provide more complete user stories and discuss them further.

Since it will eventually become openHAB's new frontend used and seen by all, while it's clear I'll probably end up making the bulk of the initial design and implementation, I feel it should not become a difficult-to-maintain one-man show, so I will try to keep the process as transparent and open as possible - and make sure it will remain as easy as possible for other developers (and eventual future maintainers) to get familiar with the code. This means if there's a new design choice or difficulty warranting a discussion before jumping into implementation I'll do my best to follow the standard workflow and bootstrap those discussions in https://github.com/openhab/openhab-webui/issues first.

Looking forward to your feedback!

**Comment by kaikreuzer** [comment]
@cdjackson & @peuter would be great to have you voting as well!
